### ISSUE TITLE
So strange the train loss, Is there a bug in loss computation?

### ISSUE THREAD

**Issue by houweidong** [qqwweee/keras-yolo3#389]
I train the yolov3 on my own dataset, and i have changed the default 80 classes to my own classes(just two classes), and i generated my own train.txt and val.txt.

When a new epoch starts, the loss is so different with the loss in the end of the previous epoch. Sometimes, it is larger than the loss in the end of the previous epoch a lot, sometimes, it is smaller than.
Just like this:
```
Epoch 27/250

...
255/270 [===========================>..] - ETA: 12s - loss: 7.5485 - acc: 0.0000e+00
256/270 [===========================>..] - ETA: 11s - loss: 7.5572 - acc: 0.0000e+00
257/270 [===========================>..] - ETA: 10s - loss: 7.5588 - acc: 0.0000e+00
258/270 [===========================>..] - ETA: 9s - loss: 7.5589 - acc: 0.0000e+00 
259/270 [===========================>..] - ETA: 8s - loss: 7.5586 - acc: 0.0000e+00
260/270 [===========================>..] - ETA: 8s - loss: 7.5569 - acc: 0.0000e+00
261/270 [============================>.] - ETA: 7s - loss: 7.5637 - acc: 0.0000e+00
262/270 [============================>.] - ETA: 6s - loss: 7.5618 - acc: 0.0000e+00
263/270 [============================>.] - ETA: 5s - loss: 7.5591 - acc: 0.0000e+00
264/270 [============================>.] - ETA: 4s - loss: 7.5609 - acc: 0.0000e+00
265/270 [============================>.] - ETA: 4s - loss: 7.5591 - acc: 0.0000e+00
266/270 [============================>.] - ETA: 3s - loss: 7.5596 - acc: 0.0000e+00
267/270 [============================>.] - ETA: 2s - loss: 7.5609 - acc: 0.0000e+00
268/270 [============================>.] - ETA: 1s - loss: 7.5611 - acc: 0.0000e+00
269/270 [============================>.] - ETA: 0s - loss: 7.5597 - acc: 0.0000e+00
270/270 [==============================] - 229s 847ms/step - loss: 7.5611 - acc: 0.0000e+00 - val_loss: 68.9214 - val_acc: 0.0000e+00
Epoch 28/250

  1/270 [..............................] - ETA: 3:41 - loss: 9.0366 - acc: 0.0000e+00
  2/270 [..............................] - ETA: 3:38 - loss: 8.2517 - acc: 0.0000e+00
  3/270 [..............................] - ETA: 3:37 - loss: 7.8761 - acc: 0.0000e+00
  4/270 [..............................] - ETA: 3:36 - loss: 8.3420 - acc: 0.0000e+00
  5/270 [..............................] - ETA: 3:35 - loss: 8.4526 - acc: 0.0000e+00
  6/270 [..............................] - ETA: 3:34 - loss: 8.1700 - acc: 0.0000e+00
  7/270 [..............................] - ETA: 3:33 - loss: 8.1300 - acc: 0.0000e+00
  8/270 [..............................] - ETA: 3:32 - loss: 8.1058 - acc: 0.0000e+00
  9/270 [>.............................] - ETA: 3:32 - loss: 8.2785 - acc: 0.0000e+00
 10/270 [>.............................] - ETA: 3:31 - loss: 8.0986 - acc: 0.0000e+00
 11/270 [>.............................] - ETA: 3:30 - loss: 8.1271 - acc: 0.0000e+00
 12/270 [>.............................] - ETA: 3:30 - loss: 8.0108 - acc: 0.0000e+00
 13/270 [>.............................] - ETA: 3:29 - loss: 8.0625 - acc: 0.0000e+00
 14/270 [>.............................] - ETA: 3:28 - loss: 8.1935 - acc: 0.0000e+00
 15/270 [>.............................] - ETA: 3:27 - loss: 8.1508 - acc: 0.0000e+00
 16/270 [>.............................] - ETA: 3:26 - loss: 8.1542 - acc: 0.0000e+00
 17/270 [>.............................] - ETA: 3:26 - loss: 8.2251 - acc: 0.0000e+00
 ...
267/270 [============================>.] - ETA: 2s - loss: 7.5793 - acc: 0.0000e+00
268/270 [============================>.] - ETA: 1s - loss: 7.5740 - acc: 0.0000e+00
269/270 [============================>.] - ETA: 0s - loss: 7.5733 - acc: 0.0000e+00
270/270 [==============================] - 229s 848ms/step - loss: 7.5695 - acc: 0.0000e+00 - val_loss: 64.9514 - val_acc: 0.0000e+00
```

Why? Are there some necessary steps i have not done when train my own dataset?

**Comment by lcltopismine** [comment]
It is possible, if you set shuffle=True, your loss should decrease after few epochs, but it's not the case here. Can you provide more information?
